src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -89,6 +91,15 @@ static bool exclusive = false;
 #endif
 #define htonll(a) ntohll(a)
 
+static int do_unmap();
+
+static void handle_signal(int signum)
+{
+  assert(signum == SIGINT || signum == SIGTERM);
+  derr << "*** Got signal " << sig_str(signum) << " ***" << dendl;
+  ioctl(nbd, NBD_DISCONNECT);
```

Comment:
I am not sure there is a valid situation when ioctl(nbd, NBD_DISCONNECT) may fail in our case. Still there may be bugs. And if it fails we are better to know about this.